qemu/tests/cgroup.py

```diff
@@ -1133,9 +1134,12 @@ def _generate_verification(cpusets, cpus):
             logging.error("Test requires at least 2 vCPUs.")
             raise exceptions.TestSkipError("Test requires at least 2 vCPUs.")
         # Check whether smp changed and recreate VM if so
-        if vm_cpus != params.get("smp", 0):
+        if vm_cpus != params.get_numeric("smp"):
             logging.info("Expected VM reload.")
             params['smp'] = vm_cpus
+            if arch.ARCH in ('ppc64', 'ppc64le'):
```

Comment:
Please refer to https://github.com/avocado-framework/avocado-vt/blob/5c26e99241669279d79bb8b5691e06e1ccf0376b/virttest/qemu_vm.py#L1658-L1706 and my above comment to set up vm_cpus, and `cgroup_use_half_smp` is used in L1127
BTW, why only ppc platforms?

Reply:
Updated it, could you help to review it again ? @PaulYuuu  Thanks.